torch_xla/csrc/tensor_util.cpp

```diff
@@ -608,21 +650,46 @@ void PopulateTensorBuffer(const at::Tensor& tensor,
 
 xla::ComputationClient::DataPtr TensorToXlaData(const at::Tensor& tensor,
                                                 const xla::Shape& shape,
-                                                const Device& device) {
-  auto populate_fn =
-      [&](const xla::ComputationClient::TensorSource& source_tensor,
-          void* dest_buffer, size_t dest_buffer_size) {
-        PopulateTensorBuffer(tensor, source_tensor.shape, dest_buffer,
-                             dest_buffer_size, device);
-      };
+                                                const Device& device,
+                                                bool transfer_async) {
+  XLA_TIMED("TensorToData");
+  if (transfer_async) {
+    std::shared_ptr<DataAsync> async = std::make_shared<DataAsync>();
+    auto populate_mwait = std::make_shared<xla::util::MultiWait>(1);
```

Comment:
can we replace the hardcoded `1` with an defined paramter for more code clarity?

Reply:
Hmm, the reason we set it to 1 is that there is only one function we expect to run async, which is the `populate_fn` below. What kind of defined paramter are you thinking about?

Other usecase of of this function are in https://github.com/pytorch/xla/blob/e0eb2a7bbc1e7dcefa4d83c88548a9dad7c3a387/third_party/xla_client/xrt_computation_client.cc#L361

Reply:
Something like this:

`#define NUM_ASYNC_FUNCTIONS 1`

Reply:
Maybe something like
```
std::make_shared<xla::util::MultiWait>(/*num_async_func=*/1)
```
?

Reply:
updated to `std::make_shared<xla::util::MultiWait>(/*num_wait=*/1);`